### DISCUSSION TITLE
Very high CPU usage with neovim's lsp client

### DISCUSSION THREAD

**Discussion by binarious** [phpactor/phpactor#2002] [Q&A] [answered]
I'm using [neovim/nvim-lspconfig's phpactor preset](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/phpactor.lua) without any special settings (`lspconfig.phpactor.setup {}`). Everything is working, but the phpactor language server is constantly eating up ~80 % CPU. 
[intelephense's preset](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/intelephense.lua) isn't using any CPU while unused and not much while actively in use.

Any ideas?

**Comment by dantleech** [comment, marked as answer]
is it indexing? can confirm that I have many instances running and all are idle. does this happen in all projects or some?

**Reply by binarious** [reply]
> is it indexing

It's not giving any output besides listening to STDIN when given `-vvv`. I don't know if it's indexing. It fails to detect some classes so that could be the case.

> does this happen in all projects or some?

Currently in all projects, but I've just configured phpactor so it might just need some time indexing. A log output regarding this would be helpful, I guess.

**Reply by binarious** [reply]
You we're right. It seems to have been a fairly long indexing process. The instances are now idle. Sorry for bothering.
